src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -130,9 +130,9 @@ public CacheTagHelper(CacheTagHelperMemoryCacheFactory factory, HtmlEncoder html
                     entry.SetOptions(options);
 
                     entry.Value = result;
-                }
 
                 tcs.SetResult(content);
+                entry.Dispose();
```

Comment:
Can you please add a comment? I want to make sure that some enterprising young buck doesn't come along later and *clean this up*